app/src/main/java/com/example/weatherapp_miniprojet/Activities/MainActivity.java

```diff
@@ -21,142 +16,179 @@
+        pre_meteo2 = findViewById(R.id.preMeteo2);
+        pre_meteo3 = findViewById(R.id.preMeteo3);
+        pre_meteo4 = findViewById(R.id.preMeteo4);
+        pre_meteo5 = findViewById(R.id.preMeteo5);
+        temp_max = findViewById(R.id.tempMax);
+        temp_max1 = findViewById(R.id.tempMax1);
+        temp_max2 = findViewById(R.id.tempMax2);
+        temp_max3 = findViewById(R.id.tempMax3);
+        temp_max4 = findViewById(R.id.tempMax4);
+        temp_max5 = findViewById(R.id.tempMax5);
+        temp_min = findViewById(R.id.tempMin);
+        temp_min1 = findViewById(R.id.tempMin1);
+        temp_min2 = findViewById(R.id.tempMin2);
+        temp_min3 = findViewById(R.id.tempMin3);
+        temp_min4 = findViewById(R.id.tempMin4);
+        temp_min5 = findViewById(R.id.tempMin5);
 
-        prev.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent prevIntent = new Intent(getApplicationContext(),PrevisionActivity.class);
-                startActivity(prevIntent);
-            }
-        });
         btnVille.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent villeIntent = new Intent(getApplicationContext(),VilleFavorie.class);
+                Intent villeIntent = new Intent(getApplicationContext(), VilleFavorie.class);
                 startActivity(villeIntent);
             }
         });
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
         // setFlags() : Définissez les indicateurs de la fenêtre.
         getSupportActionBar().hide();
 
-        StrictMode.ThreadPolicy policy= new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
         StrictMode.setThreadPolicy(policy);
         requestPermissions();
 
-        FusedLocationProviderClient mFusedLocationProviderClient;
-        // FusedLocationProviderClient : Le principal point d'entrée pour interagir avec le fournisseur d'emplacement fusionné
-        mFusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this);
-        if(ActivityCompat.checkSelfPermission(MainActivity.this, ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED){
-            // PackageManager.PERMISSION_GRANTED : Résultat du contrôle des autorisations: il est renvoyé par checkPermission si l'autorisation a été accordée au package donné
-            return;
+        //Value Gotten From VilleFavouie Activity
+        Intent intent = getIntent();
+        String getVille = intent.getStringExtra("ville");
+        if (getVille != null) {
+            cityField.setText(getVille);
+            try {
+                ArrayList<String> infos = findByVille(getVille);
+                if (!infos.equals(null)) {
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.